client/src/js/components/bhEmployeeFinder.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * The employee finder component
+ *
+ * @module components/bhEmployeeFinder
+ *
+ * @description a component to deal with employee, it let you choose a employee
+ * 
+ **/
+ 
+//An employee finder component controller
+ 
+function employeeFinderController (employeeService){ 	
+ 	var ctrl = this;
+
+ 	ctrl.searchOptions = [
+ 		{key : 'code', label : 'EMPLOYEE.CODE', placeHolder : 'EMPLOYEE.SEARCH_BY_CODE'},
+ 		{key : 'names', label : 'EMPLOYEE.NAME', placeHolder : 'EMPLOYEE.SEARCH_BY_NAME'}
+ 	];
+
+ 	function load (){
+ 		ctrl.selectedEmployee = null;
+ 		ctrl.employeeValue = null;
+ 		ctrl.session = {state : 'finding'};
+ 		setSearchOption(ctrl.searchOptions[0]); 
+
+ 		employeeService.read()
+ 		.then(function (employees){
+ 			ctrl.employees = employees;
+ 		});		
+	}
+
+	function setSearchOption (searchOption){
+		ctrl.selectedSearchOption = searchOption;
+	}
+
+	function selectEmployee (item){
+		ctrl.selectedEmployee = item;
+		ctrl.employeeValue = item.account_id;
+		ctrl.session.state = 'found';
+	}
+
+	function search (text) {
+
+		return ctrl.employees.filter(function (item){
+			var content = ctrl.selectedSearchOption.key === 'names' ? [item.prenom, item.name, item.postnom].join('') : item.code_employee;
+			return new RegExp(text, 'i').test(content); //build a regex a look for text occurence in the employee names without case sensive
+		});
+	}
+
+	function formatEmployee (employee) {
```

Comment:
Since you use this in your template and in your search function, it would be more efficient (and readable) to assign the `formatEmployee(employee)` result to each employee object as a human readable label.  You can do this on the initial data download.  It would look like this:
```js
function load() {
  employeeService.read()
  .then(function (employees) {

    // cache a human readable label
    employees.forEach(function (employee) {
      employee.label = formatEmployee(employee);
     });

     // the view now has access to the employee.label property for disaply.
     ctrl.employees = employees
  });
}
```
then, in the view, you can use
```html
<input 
  uib-typeahead="employee as employee.label for employee in $ctrl.search($viewValue)"
  />
```
This method is both more efficient  (the function is called only N times for N employees) and easier to read in the view.